### PR TITLE
fix(laravel): collection relations other than HasMany

### LINKED ISSUE #6736: [Laravel] Relation methods other than HasMany that return multiple results are not working
**API Platform version(s) affected**: 3.\*, 4.\*

**Description**  
If I use a relation other than HasMany that returns multiple items then the API does not work correctly because it tries to return a single item

**How to reproduce**  
use a HasManyThrough, BelongsToMany, MorphMany, or MorphToMany relation

**Possible Solution**  
[In src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php#L96](https://github.com/api-platform/core/blob/4171d5f9cd41731b857c53a186270ba0626baedf/src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php#L96)
Change this if statement to include other possible relations that return a collection

### PR DESCRIPTION
**Pull request by amermchaudhary**
| Q             | A
| ------------- | ---
| Branch?       | main for features / current stable version branch for bug fixes 
| Tickets       | Closes #6736 
| License       | MIT
| Doc PR        | 

Fixes the issue on collection relations other than HasMany not working for Laravel

### CODE REVIEW

**Comment by soyuka** [comment]
nice thanks! Isn't there an interface that all these `HasMany` classes implement to avoid listing them?

**Comment by amermchaudhary** [comment]
> nice thanks! Isn't there an interface that all these `HasMany` classes implement to avoid listing them?

sadly, no. All classes extend Relation class but nothing specific for `HasMany` classes

File: src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php [hunk 1]
```diff
@@ -93,7 +97,7 @@ public function create(string $resourceClass, string $property, array $options =
             }
 
             $collection = false;
-            if (HasMany::class === $relation['type']) {
+            if (\in_array($relation['type'], [HasMany::class, HasManyThrough::class, BelongsToMany::class, MorphMany::class, MorphToMany::class], true)) {
```
Comment (dunglas) [inline src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php:100]: Nit: using an associative array with types as keys and `true` as values, or just testing all conditions directly in the if (separated by `||`) will be a bit faster than using `in_array()`.

Comment (dunglas) [inline src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyMetadataFactory.php:100]: Maybe should we use a `match()` instead?

**Comment by soyuka** [comment]
awesome! thanks @amermchaudhary

**Comment by Jehong-Ahn** [comment]
I'm new to API Platform and I'm still having trouble integrating belongsToMany relationships. Could you provide an example, or perhaps enhance the examples in `/src/Laravel/workbench/app/Models`?

**Comment by amermchaudhary** [comment]
@Jehong-Ahn the issue is fixed in version 4.0.6. Please make sure you have the latest code

**Comment by amermchaudhary** [comment]
Example:
`
    public function projects() : HasManyThrough {
        return $this->hasManyThrough(Project::class, ProjectUser::class, 'user_id', 'id', 'id', 'project_id');
    }
`

**Comment by Jehong-Ahn** [comment]
Thanks @amermchaudhary! I see we can use HasManyThrough.

**Pull request merged**